### PR TITLE
Move DagFileProcessor and DagFileProcessorProcess out of scheduler_job.py

### PR DESCRIPTION
**Pull request by ephraimbuddy**
This change moves DagFileProcessor and DagFileProcessorProcess out of scheduler_job.py.
This is to help newcomers in understanding the scheduler and dag file processor as the files will
be much smaller.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/jobs/scheduler_job.py [hunk 1]
```diff
@@ -817,16 +208,16 @@ def _change_state_for_tis_without_dagrun(
         """
         tis_changed = 0
         query = (
-            session.query(models.TaskInstance)
-            .outerjoin(models.TaskInstance.dag_run)
-            .filter(models.TaskInstance.dag_id.in_(list(self.dagbag.dag_ids)))
-            .filter(models.TaskInstance.state.in_(old_states))
+            session.query(TI)
```
Comment (ashb) [inline airflow/jobs/scheduler_job.py]: I think I'd rather we kept this as `TaskInstance` or `models.TaskInstance` instead of `TI`.


But either way _this_ type of change should be in a different PR -- this is already a big PR, so keeping changes to a minimum helps the reviewer.

**Review by ashb — COMMENTED** [review]
This PR would mean we have airflow/utils/dag_processing.py and airflow/utils/dagfile_processor.py

I know it might be a big file, but what do you think of putting DagFileProcessor class etc in to the existing `airflow/utils/dag_processing.py` -- that way all of the dag parsing code lives in one file.

**Comment by ephraimbuddy** [comment]
> This PR would mean we have airflow/utils/dag_processing.py and airflow/utils/dagfile_processor.py
> 
> I know it might be a big file, but what do you think of putting DagFileProcessor class etc in to the existing `airflow/utils/dag_processing.py` -- that way all of the dag parsing code lives in one file.

It'll be big but being in the same context, I think it's worth being together. Will fix it

File: tests/jobs/test_scheduler_job.py [hunk 2]
```diff
@@ -3051,7 +2367,7 @@ def test_verify_integrity_if_dag_not_changed(self):
         dr = drs[0]
 
         # Verify that DagRun.verify_integrity is not called
-        with mock.patch('airflow.jobs.scheduler_job.DagRun.verify_integrity') as mock_verify_integrity:
+        with mock.patch('airflow.jobs.scheduler_job.DR.verify_integrity') as mock_verify_integrity:
```
Comment (ephraimbuddy) [inline tests/jobs/test_scheduler_job.py]:
```suggestion
        with mock.patch('airflow.jobs.scheduler_job.DagRun.verify_integrity') as mock_verify_integrity:
```

**Review by ashb — COMMENTED** [review]
Nice, moving code out of utils/ gets a 👍🏻 from me!

`airflow/dag_processing/dag_processing.py` is a little odd,

How about

`airflow/dag_processing/manager.py` and `airflow/dag_processing/processor.py` as file names? gives it _a_ bit more of a clue what is in each

**Comment by ephraimbuddy** [comment]
> Nice, moving code out of utils/ gets a 👍🏻 from me!
> 
> `airflow/dag_processing/dag_processing.py` is a little odd,
> 
> How about
> 
> `airflow/dag_processing/manager.py` and `airflow/dag_processing/processor.py` as file names? gives it _a_ bit more of a clue what is in each

Nice! Will do that 👍

**Review by kaxil — COMMENTED** [review]
❤️ Nice

**Review by jedcunningham — APPROVED** [review]

**Review by kaxil — APPROVED** [review]
